File: Discrete Mathematics/Fundamentals OF Mathematical Logics/PYTHON Implementations/exclusive_disjunction.py
# A function to implement exclusive disjunction

def check_exclusive_disjunction(p, q):
    if (p or q) == True and not (p and q):
        return True
    else:
        return False

result = check_exclusive_disjunction(False, True)
print(result) # output: True

result = check_exclusive_disjunction(True, True)
print(result) # output: False

result = check_exclusive_disjunction(True, False)
print(result) # output: True

result = check_exclusive_disjunction(False, False)
print(result) # output: False

File: Discrete_Mathematics/Fundamentals OF Mathematical Logics/Java Implementation/Exclusive_Disjunction.java

    // Java Implementation of Exclusive Disjunction Propositions

public class Exclusive_Disjunction {
    // Function to check Exclusive Disjunction proposition
        public boolean isExclusive_Disjunction(boolean p, boolean q) {
        if (((p || q) == true) && !(p && q == true)){
            return true;
        }
        else{
            return false;
        }

        }

    // test
    boolean output;

    public void main(String[] args) {
        output = isExclusive_Disjunction(true, true);
        System.out.println(output); // output: false

        output = isExclusive_Disjunction(false, true);
        System.out.println(output); // output: true

        output = isExclusive_Disjunction(true, false);
        System.out.println(output); // output: true

        output = isExclusive_Disjunction(false, false);
        System.out.println(output); // output: false
    }
    }

